cloudinit/net/freebsd.py

```diff
@@ -0,0 +1,164 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import re
+
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import util
+from cloudinit.distros import rhel_util
+from cloudinit.distros.parsers.resolv_conf import ResolvConf
+
+from . import renderer
+
+LOG = logging.getLogger(__name__)
+
+
+class Renderer(renderer.Renderer):
+    resolv_conf_fn = 'etc/resolv.conf'
+    rc_conf_fn = 'etc/rc.conf'
+
+    def __init__(self, config=None):
+        if not config:
+            config = {}
+        self.dhcp_interfaces = []
+        self._postcmds = config.get('postcmds', True)
+
+    def _write_ifconfig_entries(self, settings, target=None):
+        ifname_by_mac = net.get_interfaces_by_mac()
+        for interface in settings.iter_interfaces():
+            device_name = interface.get("name")
+            device_mac = interface.get("mac_address")
+            if device_name and re.match(r'^lo\d+$', device_name):
+                continue
+            if device_mac and device_name:
+                cur_name = ifname_by_mac[device_mac]
+                if not cur_name:
+                    LOG.info('Cannot find any device with MAC %s', device_mac)
+                    continue
+                if cur_name != device_name:
+                    rhel_util.update_sysconfig_file(
+                        util.target_path(target, self.rc_conf_fn), {
+                            'ifconfig_%s_name' % cur_name: device_name})
+            elif device_mac:
+                device_name = ifname_by_mac[device_mac]
```

Comment:
Same here.  I suggest refactoring the mac to name mapping into a single operation above these two blocks.

Reply:
Addressed in https://github.com/canonical/cloud-init/pull/61/commits/b49689d1a00400d39a7d2c65a6a4587490a6b3fe